gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -456,8 +486,11 @@ public ApiCallContext merge(ApiCallContext inputCallContext) {
       newDeadline = callOptions.getDeadline();
     }
 
+    boolean newIsDirectPath = grpcCallContext.isDirectPath;
+
     CallCredentials newCallCredentials = grpcCallContext.callOptions.getCredentials();
-    if (newCallCredentials == null) {
+    // If newIsDirectPath is true, then newChannel is guanrateed to be not null.
+    if (newCallCredentials == null && !newIsDirectPath) {
```

Comment:
Is this change required? I think even if `newCallCredentials` are set, is would be overridden later in the constructor? If that's the case, I would prefer to not couple the direct path logic with a merge logic in this method. This method is simply merging values of two `ApiCallContext`, how we use those values are the responsibility of the constructors.

Reply:
No it's not. This was some overlooked left-over before I moved the logic to strip the call credentials from the call options inside the private constructor. Cleaned it up now.